src/Core/Generator/SamplingModes.cs

```diff
@@ -12,18 +12,18 @@ public enum SamplingModes
         /// Represents the mode where the top N records are sampled.
         /// This mode selects a specified number of records in descending order based on a timestamp or other sorting criteria.
         /// </summary>
-        TopNSampler,
+        TopNExtractor,
```

Comment:
why the name change? pr description and design doc now out of date.

Not too big of an issue but the name changes now result in inconsistent sampler names. Two of the three have the suffix "sampler" and this one does not.

Reply:
Suggested by @sajeetharan https://github.com/Azure/data-api-builder/pull/2288#issuecomment-2316769732

Reply:
gotcha. Closing out since that's what he recommended.